ci/envs/310-oldest.yaml

```diff
@@ -13,6 +13,7 @@ dependencies:
   - pandas>=1.4.0,!=1.5.0,<2
   - shapely=2.0
   - tqdm=4.63.0
+  - numba
```

Comment:
Should we add an oldest `numba` pin here?

Reply:
Also will need to update:
* `pyproject.toml`
* `environment.yml`
* anything else?